tools/pythonpkg/pyduckdb/protocols.py

```diff
@@ -0,0 +1,6 @@
+from typing import Protocol, runtime_checkable
+
+@runtime_checkable
+class Placeholder(Protocol):
+    def read(self):
+        pass
```

Comment:
Just a clarification question, how is this being currently used?

Reply:
Yea that's partly the problem with this PR, it's a great idea, but it has no use at the moment, because either the candidate that would apply - doesn't meet certain requirements, or the actual thing we're trying to ducktype does not support duck typing (Arrow..)